### PR TITLE
21 backend signup

### PR DESCRIPTION
**Pull request by momenmarwan**

### CODE REVIEW

**Review by momenqudaih — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -5,38 +5,42 @@
...
     "start": "cross-env NODE_ENV=production node ./dist/index.js",
     "lint": "cross-env eslint . --ext .ts",
     "lint-fix": "cross-env eslint . --ext .ts --fix",
     "dev": "cross-env NODE_ENV=development npx nodemon server/index.ts",
     "prepare": "husky install",
-    "db-build":"cross-env NODE_ENV=development ts-node server/db/build.ts",
-    "seed-build":"cross-env NODE_ENV=development ts-node server/db/seed.ts"
-
+    "db-build": "cross-env NODE_ENV=development ts-node server/db/build.ts",
+    "seed-build": "cross-env NODE_ENV=development npm run db-build && ts-node server/db/seed.ts"
```
Comment (moustf) [inline package.json:16]: Remove the `npm run db-build &&` from here.

File: server/controllers/auth/signup.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    const token = await signToken({
+      id: created.id as number,
+      username,
+      role: roleId,
+    });
+
+    return res.status(StatusCodes.Created).cookie('token', token).json({
+      error: false,
+      msg: 'The user have been created successfully!',
+      data: created,
```
Comment (moustf) [inline server/controllers/auth/signup.ts:46]: Don't return the whole data, id and user name are more than enough.
This user data includes the password, and it'll be wrong to return it.

File: server/routes/auth.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { Router } from 'express';
+import { signupController } from '../controllers';
+
+const authRouter: Router = Router();
```
Comment (moustf) [inline server/routes/auth.ts:4]: Remove the Router type from here.

**Review by Farah202324 — APPROVED** [review]

**Review by momenqudaih — APPROVED** [review]
good job momen

**Review by moustf — APPROVED** [review]

**Pull request merged**